sigma/plugins.py

```diff
@@ -180,6 +206,62 @@ def get_pipeline_resolver(self) -> ProcessingPipelineResolver:
             }
         )
 
+    @staticmethod
+    def _get_backend_identifier(obj: Any, default: str) -> Optional[str]:
+        """
+        Get the identifier of a backend object. This is either the identifier attribute of
+        the object, the __identifier__ attribute of the object, or the __class__ attribute
+        of the object. The identifier is then converted to snake_case. If the identifier is
+        empty, the default is returned.
+
+        Args:
+            obj: The Backend object to get the identifier from.
+            default: The default identifier to return if no identifier could be found.
+
+        Returns:
+            The identifier of the backend object in snake_case or the default identifier.
+        """
+
+        def removesuffix(base: str, suffix: str) -> str:
+            """Removes the suffix from the string if it exists.
+            This is a backport of the Python 3.9 removesuffix method.
+            """
+            if base.endswith(suffix):
+                return base[: len(base) - len(suffix)]
+            return base
```

Comment:
This is introduced in Python 3.9, which is why tests failed on earlier commits when running on Python 3.8.